src/mo_reader.cpp

```diff
@@ -85,13 +85,16 @@ uint32_t mo_file_reader::read_section() {
 	return ret;
 }
 
-
 std::string mo_file_reader::get_system_lang_short_name() const {
 #ifndef _WIN32
-	std::string lang_env = std::getenv("LANG");
+	char *env = std::getenv("LANG");
+	std::string lang_env;
+	if (env != nullptr)
+		lang_env = std::getenv("LANG");
```

Comment:
Now it checks against nullptr, but still:

1) getenv() should be called just 1 time here, inside if do lang_env = string(env)
2) getenv is unsafe function (not thread safe).

So I would propose that instead, we should write simple, secure, singleton class that on begining of int main() will copy all env variables and then provide a secure interfaces to access this values (well their copies, ignoring any changes setenv in program).

As in http://stackoverflow.com/questions/2085302/printing-all-environment-variables-in-c-c